arcgis-ios-sdk-samples/Layers/Group layers/LayersTableViewController.swift

```diff
@@ -73,6 +111,18 @@ class LayersTableViewController: UITableViewController, GroupLayersCellDelegate,
         return headerView
     }
     
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let groupLayer = layers[indexPath.section] as! AGSGroupLayer
+        let childLayers = groupLayer.layers as! [AGSLayer]
+        
+        var indexPathsToReload = [indexPath]
+        if let indexOfPreviouslyVisibleLayer = childLayers.firstIndex(where: { $0.isVisible }) {
+            indexPathsToReload.append(IndexPath(row: indexOfPreviouslyVisibleLayer, section: indexPath.section))
+        }
+        childLayers[indexPath.row].isVisible = true
+        tableView.reloadRows(at: indexPathsToReload, with: .automatic)
```

Comment:
I had forgot that reloading parts of a table view doesn't play nicely with observing the content size, for some reason. My first idea was that the observation could be removed as soon as a change was observed, but the first observed change isn't the correct size, so that wouldn't work. My other thought would be to have a custom `func reloadRows(at indexPaths: [IndexPath])` function that first removes the content size observation (if it still exists) and then calls `reloadRows` on the table view. What do you think of that idea?

Reply:
I think that's a good idea. I gave it a try, please let me know what you think!